tools/repoUpdateScripts/updateAzureDataStudio.ps1

```diff
@@ -0,0 +1,274 @@
+
+                'EndObject'
+                {
+                    if ($foundPowerShell -and $depth -eq $startDepth + 1)
+                    {
+                        return @{
+                            Start = @{
+                                Line = $startLine
+                                Column = $startColumn
+                            }
+                            End = @{
+                                Line = $jsonReader.LineNumber
+                                Column = $jsonReader.LinePosition
+                            }
+                        }
+                    }
+                    $depth--
+                    continue
+                }
+
+                'PropertyName'
+                {
+                    if ($jsonReader.Value -eq 'extensionName')
+                    {
+                        $awaitingExtensionName = $true
+                    }
+                    continue
+                }
+
+                'String'
+                {
+                    if (-not $awaitingExtensionName)
+                    {
+                        continue
+                    }
+
+                    $awaitingExtensionName = $false
+
+                    if ($jsonReader.Value -eq 'PowerShell')
+                    {
+                        $foundPowerShell = $true
+                    }
+
+                    continue
+                }
+            }
+        }
+    }
+    finally
+    {
+        $reader.Dispose()
+        $jsonReader.Dispose()
+    }
+
+    throw 'Did not find PowerShell extension'
+}
+
+function UpdateGalleryFile
+{
+    param(
+        [Parameter(Mandatory)]
+        [version]
+        $ExtensionVersion,
+
+        [Parameter()]
+        [string]
+        $GalleryFilePath = './extensionsGallery-insider.json'
+    )
+
+    # Create a new PowerShell extension entry
+    $powershellEntry = NewPowerShellExtensionEntry -ExtensionVersion $ExtensionVersion
+    $entryStr = ConvertToIndentedJson $powershellEntry -IndentChar "`t" -IndentWidth 1
+
+    # Find the position in the existing file where the PowerShell extension should go
+    $galleryFileContent = Get-Content -Raw $GalleryFilePath
+    $span = FindPSExtensionJsonSpan -GalleryExtensionFileContent $galleryFileContent
+    $startOffset = GetStringOffsetFromSpan -String $galleryFileContent -EndLine $span.Start.Line -Column $span.Start.Column
+    $endOffset = GetStringOffsetFromSpan -String $galleryFileContent -EndLine $span.End.Line -StartLine $span.Start.Line -Column $span.End.Column -InitialOffset $startOffset
+
+    # Create the new file contents with the inserted segment
+    $newGalleryFileContent = ReplaceStringSegment -String $galleryFileContent -NewSegment $entryStr -StartIndex $startOffset -EndIndex ($endOffset+1) -AutoIndent
+
+    # Write out the new entry
+    SetFileContent $GalleryFilePath $newGalleryFileContent
+}
+
+$repoLocation = Join-Path ([System.IO.Path]::GetTempPath()) 'ads-temp-checkout'
+
+if (-not $BranchName)
```

Comment:
Can you put this in the default parameters?

Reply:
It depends on another parameter. Does that work?

Reply:
Ah, experimenting shows this depends on the order of parameters

Reply:
Turns out there was complicated file-reading logic that needed to be done when `$NewVersion` wasn't provided, so I had to move this back out